File: backend/src/main/java/com/_bet/service/auth/AuthService.java
package com._bet.service.auth;

import com._bet.dto.request.LoginRequest;
import com._bet.dto.request.RegistroRequest;
import com._bet.dto.response.JwtResponse;
import com._bet.dto.response.UsuarioResponse;

public interface AuthService {
    
    /**
     * Registra un nuevo usuario en el sistema
     * @param registroRequest Datos del usuario a registrar
     * @return UsuarioResponse con los datos del usuario registrado
     * @throws RuntimeException si el usuario ya existe o hay errores de validación
     */
    UsuarioResponse registrar(RegistroRequest registroRequest);
    
    /**
     * Autentica un usuario y genera un token JWT
     * @param loginRequest Credenciales del usuario (username/email y password)
     * @return JwtResponse con el token y datos del usuario
     * @throws RuntimeException si las credenciales son incorrectas
     */
    JwtResponse login(LoginRequest loginRequest);
}

File: frontend/mobile/src/service/authService.ts
import { api } from './apiBase';
import {
  LoginRequest,
  RegistroRequest,
  JwtResponse,
  UsuarioResponse,
  ApiResponseWrapper
} from '../types/authTypes';

const BASE_URL = '/auth';

export const authService = {
  registro: async (registroRequest: RegistroRequest): Promise<ApiResponseWrapper<UsuarioResponse>> => {
    const response = await api.post<UsuarioResponse>(`${BASE_URL}/registro`, registroRequest);
    return response;
  },

  login: async (loginRequest: LoginRequest): Promise<ApiResponseWrapper<JwtResponse>> => {
    return await api.post<JwtResponse>(`${BASE_URL}/login`, loginRequest);
  }
};
